crates/milli/src/update/settings.rs

```diff
@@ -1418,6 +1422,106 @@ impl<'a, 't, 'i> Settings<'a, 't, 'i> {
 
         Ok(())
     }
+
+    pub fn execute<'indexer, MSP>(
+        mut self,
+        must_stop_processing: &'indexer MSP,
+        progress: &'indexer Progress,
+    ) -> Result<Option<ChannelCongestion>>
+    where
+        MSP: Fn() -> bool + Sync,
+    {
+        // force the old indexer if the environment says so
+        if std::env::var_os("MEILI_EXPERIMENTAL_NO_EDITION_2024_FOR_SETTINGS").is_some() {
```

Comment:
Sorry for bringing this when I'm the one who originally set to get the variable here, but I think it would be best if it was a function parameter, and if the env variable could be fetch from the meilisearch crate.

1. cleaner from a separation of concerns perspective (milli doesn't read the env variable itself)
2. makes it easier to test both setting indexers